crates/pathfinder/src/storage/state.rs

```diff
@@ -396,6 +396,32 @@ impl StarknetBlocksTable {
         }
     }
 
+    /// Returns the [hash](StarknetBlockHash) and [number](StarknetBlockNumber) of the latest block.
+    pub fn get_latest_hash_and_number(
+        tx: &Transaction<'_>,
+    ) -> anyhow::Result<Option<(StarknetBlockHash, StarknetBlockNumber)>> {
+        let mut statement =
+            tx.prepare("SELECT hash, number FROM starknet_blocks ORDER BY number DESC LIMIT 1")?;
+        let mut rows = statement.query([])?;
+        let row = rows.next().context("Iterate rows")?;
+        match row {
+            Some(row) => {
+                let hash = row
+                    .get_ref_unwrap("hash")
+                    .as_blob()
+                    .expect("hash column should exist");
+                let hash =
+                    StarkHash::from_be_slice(hash).expect("hash column should contain valid hash");
+                let hash = StarknetBlockHash(hash);
+
+                let number = row.get_ref_unwrap("number").as_i64().unwrap() as u64;
+                let number = StarknetBlockNumber(number);
+                Ok(Some((hash, number)))
+            }
+            None => Ok(None),
+        }
```

Comment:
```suggestion
        let maybe =
            tx.query_row(
                "SELECT hash, number FROM starknet_blocks ORDER BY number DESC LIMIT 1", 
                [], 
                |row| {
                    let hash = row.get_ref_unwrap(0).as_blob().expect("...");
                    let hash = StarkHash::from_be_slice(hash).expect("...");
                    let hash = StarknetBlockHash(hash);
                    let num = row.get_ref_unwrap(1).as_i64().expect("...") as u64;
                    let num = StarknetBlockNumber(num);
                    Ok((hash, num))
                }).optional()?;
        Ok(maybe)
```

I query_row kind would suffice since it's just less code. Maybe prepared statements make more sense when we are executing the single prepared many times?

Since we don't use the sqlite in strict mode it's still possible to hit those `as_blob()` InvalidColumnType's as one can write a string for example to some row and getting it as blob would fail. Similarly it could be an int.